### PR TITLE
[-e] CLI implementation to create editable package (and remove link)

### LINKED ISSUE #3903: [-e] Define cli syntax for packages in editable mode
I'm working in a new feature: **install packages in editable mode.** ([branch](https://github.com/conan-io/conan/projects/1))

This feature should implement _the same_ behavior as the Python one: `pip install -e .` that makes the package in the working copy available system wide so you can use it from the sources you are editing.

In Conan we would like to allow the user to install a package (a full reference) in editable mode, so every recipe making use of it will use the recipe, includes, binaries,... from the working copy. This simplifies the workflow when dealing with several interdependent libraries at the same time. Changes in a package will be taken into account in the dependent one without calling any Conan command (of course, build/link is still needed to compile and update `.lib` or `.bin` files).

Ideas:

```
$ conan editable <path_to_conanfile.py> package/version@user/channel
$ conan install -e <path_to_conanfile.py> package/version@user/channel
```

but I want feedback from every @conan-io/barbarians and @conan-io/community-leaders, and everyone interested, to design a new command or use an existing one. We should also think about how this new option/command interferes with existing ones.

With the first implementation, I will provide a working example and explain what the package has to satisfy in order to allow _editable mode_. 💪 

Thanks for your feedback.

### PR DESCRIPTION
**Pull request by jgsogo**
Changelog: Feature: Add install argument `--editable` to link a working copy with a reference.

- [x] closes #3903 
- [ ] Need docs
- [x] Add tests creating packages

Build on top of #3804 and #3902 (changes isolated here: https://github.com/jgsogo/conan/pull/6)

### CODE REVIEW

File: conans/client/graph/graph_binaries.py [hunk 1]
```diff
@@ -114,8 +132,11 @@ def _evaluate_node(self, node, build_mode, update, evaluated_references, remote_
                 else:
                     output.warn("Can't update, no remote defined")
             if not node.binary:
-                node.binary = BINARY_CACHE
-                package_hash = ConanInfo.load_from_package(package_folder).recipe_hash
+                if node.recipe == RECIPE_EDITABLE:
+                    node.binary == BINARY_EDITABLE
```
Comment (jgsogo) [inline conans/client/graph/graph_binaries.py]: BUG!!!

File: conans/test/command/editable_reference/header_only_test.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        base_folder = tempfile.mkdtemp(suffix='conans', dir=CONAN_TEST_FOLDER)
+
+        # Editable project
+        client_editable = HeaderOnlyLibTestClient(base_folder=base_folder)
+        client_editable.make_editable(full_reference="MyLib/0.1@user/editable")
+        # client_editable.run("editable . MyLib/0.1@user/editable")  # 'Install' as editable
+
+        # Consumer project
+        client = TestClient(base_folder=base_folder)
+        conanfile_txt = """
```
Comment (danimtb) [inline conans/test/command/editable_reference/header_only_test.py]: This is not the content of a *conanfile.txt*

File: conans/test/command/editable_reference/header_only_test.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    def make_editable(self, full_reference):
+        conan_ref = ConanFileReference.loads(full_reference)
+        cache_dir = self.client_cache.conan(conan_ref)
+        save(os.path.join(cache_dir, LINKED_FOLDER_SENTINEL), content=self.current_folder)
+
+
+class EditableReferenceTest(unittest.TestCase):
+
+    def test_header_only(self):
+        # We need two clients sharing the same Conan cache
```
Comment (danimtb) [inline conans/test/command/editable_reference/header_only_test.py]: I don't understand the purpose of this test

File: conans/test/command/editable_reference/info_and_inspect_test.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        self.t.run('info .')
+        self.assertIn("    Requires:\n        parent/version@user/name", self.t.out)
+
+    def test_info_reference(self):
+        self.t.run('info {}'.format(self.reference))
+        self.assertIn("    Requires:\n        parent/version@user/name", self.t.out)
+
+    def test_inspect_path(self):
+        self.t.run('inspect .')
+        self.assertTrue(True)  # To be sure it runs
```
Comment (danimtb) [inline conans/test/command/editable_reference/info_and_inspect_test.py]: functions without asserts do not run? Anyway, an assert of the output keeping the comment will be more clear

File: conans/test/command/editable_reference/create_editable_package_test.py [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+        t = TestClient()
+        t.save(files={'conanfile.py': textwrap.dedent("""\
+            from conans import ConanFile
+            
+            class Pck(ConanFile):
+                name = "lib"
+                version = "version"
+            """)})
+        t.run('export  . {}'.format(reference))
+        t.run('install --editable=. wrong/version@user/channel'.format(reference), ignore_error=True)
```
[the remark this answers is not in the document]
Comment (jgsogo) [reply conans/test/command/editable_reference/create_editable_package_test.py]: It is valid in this branch, it hasn't been merged here yet. I'll remove the `.format(reference)`, the `assert_error` will fail when updated from `develop`.

**Pull request closed without merging**